en/engine/event/event-emit.md

```diff
@@ -0,0 +1,150 @@
+
+## Event delivery
+
+Events launched by the `dispatchEvent` method mentioned above would enter the event delivery stage. In Cocos Creator's event delivery system, we use bubble delivery. Bubble delivery will pass the event from the initiating node continually on to its parent node,  until the root node is reached or an interrupt `event.propagationStopped = true` is made in the response function of a node.
+
+![bubble-event](bubble-event.png)
+
+As shown in the picture above, when we send the event `“foobar”` from node c, if both node a and b listen to the event `“foobar”`, the event will pass to node b and a from c. For example:
+
+```ts
+// In the component script of node c
+this.node.dispatchEvent( new Event.EventCustom('foobar', true) );
+```
+
+If we want to stop the event delivery after node b intercepts the event, we can call the function `event.propagationStopped = true` to do this. Detailed methods are as follows:
+
+```ts
+// In the component script of node b
+this.node.on('foobar', (event: EventCustom) => {
+  event.propagationStopped = true;
+});
+```
+
+Be noted, when you want to dispatch a custom event, please do not use `cc.Event` because it's an abstract class, instead, you should use `cc.Event.EventCustom` to dispatch a custom event.
+
+## Event object
+
+In the call-back of the event listener, the developer will receive an event object event of the cc.Event type. `propagationStopped` is the standard API of cc.Event, other important API include:
+
+| API name                 | type             | meaning             |
+| :-------------             | :----------            |   :----------        |
+| type           | String   | type of the event (event name)                      |
+| target          | cc.Node | primary object received by the event                      |
+| currentTarget          | cc.Node | current object receiving the event; current object of the event in the bubble stage may be different from the primary object                     |
+| getType      | Function   | get the type of the event                   |
+| propagationStopped   | Boolean   | stop the bubble stage, the event will no longer pass on to the parent node while the rest of the listeners of the current node will still receive the event                     |
+| propagationImmediateStopped              | Boolean   | stop delivering the event. The event will not pass on to the parent node and the rest of the listeners of the current node。                      |
+| detail             | Function | custom event information (belongs to cc.Event.EventCustom)    |
+| setUserData             | Function | set custom event information (belongs to cc.Event.EventCustom)    |
+| getUserData             | Function | get custom event information (belongs to cc.Event.EventCustom)    |
+
+You can refer to the cc.Event and API files of its child category for a complete API list.
```

Comment:
You can refer to the cc.Event and API files of its child category for a complete API list.
-->
Please refer to the `cc.Event` and API files of its child category for a complete API list.